src/internal/digitalAsset.ts

```diff
@@ -181,3 +196,131 @@ export async function getTokenActivity(args: {
 
   return data.token_activities_v2;
 }
+
+export interface CreateCollectionOptions {
+  maxSupply?: AnyNumber;
+  mutableDescription?: boolean;
+  mutableRoyalty?: boolean;
+  mutableURI?: boolean;
+  mutableTokenDescription?: boolean;
+  mutableTokenName?: boolean;
+  mutableTokenProperties?: boolean;
+  mutableTokenURI?: boolean;
+  tokensBurnableByCreator?: boolean;
+  tokensFreezableByCreator?: boolean;
+  royaltyNumerator?: number;
+  royaltyDenominator?: number;
+}
+
+export async function createCollectionTransaction(
+  args: {
+    aptosConfig: AptosConfig;
+    creator: Account;
+    description: string;
+    name: string;
+    uri: string;
+    options?: GenerateTransactionOptions;
+  } & CreateCollectionOptions,
+): Promise<SingleSignerTransaction> {
+  const { aptosConfig, options, creator } = args;
+  const transaction = await generateTransaction({
+    aptosConfig,
+    sender: creator.accountAddress.toString(),
+    data: {
+      function: "0x4::aptos_token::create_collection",
+      arguments: [
+        // Do not change the order
+        new MoveString(args.description),
+        new U64(args.maxSupply ?? MAX_U64_BIG_INT),
+        new MoveString(args.name),
+        new MoveString(args.uri),
+        new Bool(args.mutableDescription ?? true),
+        new Bool(args.mutableRoyalty ?? true),
+        new Bool(args.mutableURI ?? true),
+        new Bool(args.mutableTokenDescription ?? true),
+        new Bool(args.mutableTokenName ?? true),
+        new Bool(args.mutableTokenProperties ?? true),
+        new Bool(args.mutableTokenURI ?? true),
+        new Bool(args.tokensBurnableByCreator ?? true),
+        new Bool(args.tokensFreezableByCreator ?? true),
+        new U64(args.royaltyNumerator ?? 0),
+        new U64(args.royaltyDenominator ?? 1),
+      ],
+    },
+    options,
+  });
+  return transaction as SingleSignerTransaction;
+}
+
+/**
+ * Queries data of a specific collection by the collection creator address and the collection name.
+ *
+ * if, for some reason, a creator account has 2 collections with the same name in v1 and v2,
+ * can pass an optional `tokenStandard` parameter to query a specific standard
+ *
+ * @param creatorAddress the address of the collection's creator
+ * @param collectionName the name of the collection
+ * @returns GetCollectionDataResponse response type
+ */
+export async function getCollectionData(args: {
+  aptosConfig: AptosConfig;
+  creatorAddress: HexInput;
+  collectionName: string;
+  options?: {
+    tokenStandard?: TokenStandard;
```

Comment:
I wish we could just ignore it and support only v2

Reply:
defaulted the arg to v2 at least